analyzers/src/SonarAnalyzer.CFG/CfgSerializer/CfgSerializer.RoslynCfgWalker.cs

```diff
@@ -106,11 +106,25 @@ private void WriteNode(BasicBlock block)
                 operation == null ? Enumerable.Empty<string>() : new[] { "## BranchValue ##" }.Concat(SerializeOperation(operation));
 
             private static IEnumerable<string> SerializeOperation(IOperation operation) =>
-                SerializeOperation(0, operation).Concat(new[] { "##########" });
+                SerializeOperation(0, null, operation).Concat(new[] { "##########" });
 
-            private static IEnumerable<string> SerializeOperation(int level, IOperation operation) =>
-                new[] { $"{level}# {operation.Serialize()}" }
-                .Concat(operation.ToSonar().Children.SelectMany(x => SerializeOperation(level + 1, x)));
+            private static IEnumerable<string> SerializeOperation(int level, string prefix, IOperation operation)
+            {
+                var ret = new List<string> { $"{level}#: {prefix}{operation.Serialize()}" };
+                var map = new Dictionary<IOperation, string>();
+                foreach (var pi in operation.GetType().GetProperties())
+                {
+                    if (pi.GetValue(operation) is IOperation value)
+                    {
+                        map[value] = map.TryGetValue(value, out var old) ? $"{old}, {pi.Name}" : pi.Name;
+                    }
+                }
+                foreach (var child in operation.ToSonar().Children)
+                {
+                    ret.AddRange(SerializeOperation(level + 1, map.TryGetValue(child, out var childPrefix) ? $"{level}#.{childPrefix}: " : null, child));
+                }
+                return ret;
+            }
```

Comment:
I find LINQ expressions easier to understand in this case. What do you think?
```suggestion
            private static IEnumerable<string> SerializeOperation(int level, string prefix, IOperation operation)
            {
                var prefixes = operation.GetType().GetProperties().GroupBy(pi => pi.GetValue(operation) as IOperation, pi => pi.Name)
                    .ToDictionary(x => x.Key, x => string.Join(", ", x));
                return new List<string> { $"{level}#: {prefix}{operation.Serialize()}" }
                    .Concat(operation.ToSonar().Children.SelectMany(child =>
                        SerializeOperation(level + 1, prefixes.TryGetValue(child, out var childPrefix) ? $"{level}#.{childPrefix}: " : null, child)));
            }
```

Reply:
I did it for the prefixes. The 2nd half has too much nesting levels and is hard to read as linq for me.